.github/scripts/check-folders.js

```diff
@@ -83,17 +84,27 @@ async function main() {
       const job_optional = optionalTestYaml['jobs'][key]
       const matrix_optional =
         Boolean(job_optional) && Boolean(job_optional.strategy) ? job_optional.strategy.matrix : {}
+
       const folders_optional = Object.keys(matrix_optional)
         .filter((key) => {
+          // These are in optional-test.yaml, under the `platforms` directory
           return !keysToIgnore.includes(key)
         })
-        .reduce((acc, key) => acc.concat(...matrix_optional[key]), [])
+        .reduce((acc, key) => {
+          return acc.concat(...matrix_optional[key])
+        }, [])
+
+      // These are in optional-test.yaml, under the `platforms` directory
+      // But the names don't match the directory name
+      // Example `platforms-m1-office` is in `platforms` directory
+      // So this renames the key so it's correctly processed
+      if (key === 'platforms-m1-office' || key === 'platforms-codesandbox') {
+        key = 'platforms'
+      }
```

Comment:
We already have a top folder named `platforms`

Reply:
Ok, then I didn't understand the hack 😅 Would you mind explaining it to me again?

Reply:
I also don't like it that much but would prefer to change that later.
Currently, the setup expects to have a 1:1 with top level directories and job names in the Actions yaml.

So the names `platforms-m1-office` and `platforms-codesandbox` are actually defined in `optional-test.yaml` but since the name is different from the directory tree it would fail the expectations.

I made codesandbox a job of its own because the test requires Puppeteer and there is no way to make the setup of Puppeteer conditional, which makes all the tests not using Puppeteer slower.

Reply:
Ah ok, gotcha, thanks for the explanation. If I understood correctly, then I'd prefer having the projects being like
- `platforms/platforms-m1-office`
- `platforms/platforms-codesandbox`

No strong opinion though :P

Reply:
That was not an option since I wanted to have codesandbox and the m1 test running separately in a different job 
not sure if that's really worth it in optional yaml at this point
I'll move them together again later in optional yaml to simplify things!